spec/helpers/react_helper_spec.rb

```diff
@@ -38,6 +39,54 @@
       end
     end
 
+    context 'when alert is present and has text and title' do
+      let(:alert) { instance_double(Alert, text: 'alert_title', title: 'alert_title') }
+
+      it 'includes alert and has text and title' do
+        helper.search_results_layout(search, {}, vertical, affiliate)
+        expect(helper).to have_received(:react_component).with(
+          'SearchResultsLayout',
+          hash_excluding(:alert)
+        )
+      end
+    end
+
+    context 'when alert is present but text is blank' do
+      let(:alert) { instance_double(Alert, text: '', title: 'alert_title') }
+
+      it 'excludes alert from data' do
+        helper.search_results_layout(search, {}, vertical, affiliate)
+        expect(helper).to have_received(:react_component).with(
+          'SearchResultsLayout',
+          hash_excluding(:alert)
+        )
+      end
+    end
+
+    context 'when alert is present but title is blank' do
+      let(:alert) { instance_double(Alert, text: 'alert_title', title: '') }
+
+      it 'excludes alert from data' do
+        helper.search_results_layout(search, {}, vertical, affiliate)
+        expect(helper).to have_received(:react_component).with(
+          'SearchResultsLayout',
+          hash_excluding(:alert)
+        )
+      end
+    end
+
+    context 'when alert is not present' do
+      let(:alert) { nil }
```

Comment:
This doesn't ensure an affiliate doesn't has an alert. This should be done with something like

```ruby
  before do
    affiliate.alert = nil
  end
```